sdks/java/io/elasticsearch/src/main/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIO.java

```diff
@@ -398,10 +394,16 @@ public Read withBatchSize(long batchSize) {
 
     @Override
     public void validate(PipelineOptions options) {
+      ConnectionConfiguration connectionConfiguration = getConnectionConfiguration();
       checkState(
```

Comment:
+1 with @echauchot: `checkArgument()` and `checkState()` should be always preferred.